internal/services/oracle/oracle_autonomous_database_backup_resource_test.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	return pointer.To(backup != nil), nil
+}
+
+func findBackupByName(ctx context.Context, client *autonomousdatabasebackups.AutonomousDatabaseBackupsClient, adbId autonomousdatabases.AutonomousDatabaseId, backupId autonomousdatabasebackups.AutonomousDatabaseBackupId) (*autonomousdatabasebackups.AutonomousDatabaseBackup, error) {
+	resp, err := client.ListByParent(ctx, autonomousdatabasebackups.AutonomousDatabaseId(adbId))
+	if err != nil {
+		return nil, fmt.Errorf("listing backups for %s: %+v", adbId.ID(), err)
+	}
+
+	id := backupId.ID()
+
+	if model := resp.Model; model != nil {
+		for _, backup := range *model {
+			if backup.Id != nil && strings.EqualFold(*backup.Id, id) {
+				return &backup, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+func (a AutonomousDatabaseBackupResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_oracle_autonomous_database_backup" "test" {
+  name                     = "backup%[2]d"
+  autonomous_database_id   = azurerm_oracle_autonomous_database.test.id
+  retention_period_in_days = 120
+  backup_type              = "LongTerm"
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a AutonomousDatabaseBackupResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_oracle_autonomous_database_backup" "test" {
+  name                     = "backup%[2]d"
+  autonomous_database_id   = azurerm_oracle_autonomous_database.test.id
+  retention_period_in_days = 120
+  backup_type              = "LongTerm"
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a AutonomousDatabaseBackupResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_oracle_autonomous_database_backup" "test" {
+  name                     = "backup%[2]d"
+  autonomous_database_id   = azurerm_oracle_autonomous_database.test.id
+  retention_period_in_days = 160
+  backup_type              = "LongTerm"
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a AutonomousDatabaseBackupResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_oracle_autonomous_database_backup" "import" {
+  name                     = azurerm_oracle_autonomous_database_backup.test.name
+  autonomous_database_id   = azurerm_oracle_autonomous_database_backup.test.autonomous_database_id
+  retention_period_in_days = azurerm_oracle_autonomous_database_backup.test.retention_period_in_days
+}
+`, a.basic(data))
+}
+
+func (a AutonomousDatabaseBackupResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
```

Comment:
Could we get a indication of what resource is being tested in the resource group name?
```suggestion
  name     = "acctestRG-oadbb-%[1]d"
```